types/responses.ts

```diff
@@ -87,6 +87,41 @@ export interface GenerateContentResponse {
   usageMetadata?: UsageMetadata;
 }
 
+/**
+ * Logprobs Result
+ * @public
+ */
+export interface LogprobsResult {
+  /** Length = total number of decoding steps. */
+  topCandidates: TopCandidates[];
+  /** 
+   * Length = total number of decoding steps. 
+   * The chosen candidates may or may not be in topCandidates.
+   */
+  chosenCandidates: Candidate[];
+}
+
+/**
+ * Candidate for the logprobs token and score.
+ * @public
+ */
+export interface Candidate {
```

Comment:
Can we name this more specifically? We already have `GenerateContentCandidate` which is the most common type of `candidate` most users will interact with. Maybe `LogProbsCandidate`?